backend-lib/src/main/rust/lib.rs

```diff
@@ -103,6 +93,13 @@ fn block_db(env: &JNIEnv<'_>, fsblockdb_root: JString<'_>) -> Result<FsBlockDb,
         .map_err(|e| format_err!("Error opening block source database connection: {:?}", e))
 }
 
+fn account_id_from_jint(account: jint) -> Result<AccountId, failure::Error> {
+    u32::try_from(account)
+        .map_err(|_| ())
+        .and_then(|id| AccountId::try_from(id).map_err(|_| ()))
```

Comment:
This `try_from` should never be able to fail, because the `u32::try_from` will have ensured that the resulting u32 falls into the valid range of account IDs. I think that using `map` and `expect` here would actually be appropriate, because if the inner `try_from` fails that _should_ indicate that something is crashingly wrong.